compiler/test/dotty/tools/debug/DebugTests.scala

```diff
@@ -0,0 +1,134 @@
+package dotty.tools.debug
+
+import com.sun.jdi.*
+import dotty.Properties
+import dotty.tools.dotc.reporting.TestReporter
+import dotty.tools.io.JFile
+import dotty.tools.vulpix.*
+import org.junit.Test
+
+import java.util.concurrent.TimeoutException
+import scala.concurrent.duration.*
+import scala.util.control.NonFatal
+
+class DebugTests:
+  import DebugTests.*
+  @Test def debug: Unit =
+    implicit val testGroup: TestGroup = TestGroup("debug")
+    // compileFile("tests/debug/eval-local-class-in-value-class.scala", TestConfiguration.defaultOptions).checkDebug()
+    compileFilesInDir("tests/debug", TestConfiguration.defaultOptions).checkDebug()
+
+object DebugTests extends ParallelTesting:
+  def maxDuration =
+    // Increase the timeout when the user is debugging the tests
+    if isUserDebugging then 3.hours else 45.seconds
+  def numberOfSlaves = Runtime.getRuntime().availableProcessors()
+  def safeMode = Properties.testsSafeMode
+  def isInteractive = SummaryReport.isInteractive
+  def testFilter = Properties.testsFilter
+  def updateCheckFiles: Boolean = Properties.testsUpdateCheckfile
+  def failedTests = TestReporter.lastRunFailedTests
+  override def debugMode = true
+
+  implicit val summaryReport: SummaryReporting = new SummaryReport
+
+  extension (test: CompilationTest)
+    private def checkDebug()(implicit summaryReport: SummaryReporting): test.type =
+      import test.*
+      checkPass(new DebugTest(targets, times, threadLimit, shouldFail || shouldSuppressOutput), "Debug")
+
+  private final class DebugTest(testSources: List[TestSource], times: Int, threadLimit: Option[Int], suppressAllOutput: Boolean)(implicit summaryReport: SummaryReporting)
+    extends RunTest(testSources, times, threadLimit, suppressAllOutput):
+
+    override def onSuccess(testSource: TestSource, reporters: Seq[TestReporter], logger: LoggedRunnable) =
+      verifyDebug(testSource.outDir, testSource, countWarnings(reporters), reporters, logger)
+
+    private def verifyDebug(dir: JFile, testSource: TestSource, warnings: Int, reporters: Seq[TestReporter], logger: LoggedRunnable) =
+      if Properties.testsNoRun then addNoRunWarning()
+      else
+        val checkFile = testSource.checkFile.getOrElse(throw new Exception("Missing check file")).toPath
+        val debugSteps = DebugStepAssert.parseCheckFile(checkFile)
+        val expressionEvaluator =
+          ExpressionEvaluator(testSource.sourceFiles, testSource.flags, testSource.runClassPath, testSource.outDir)
+        try debugMain(testSource.runClassPath): debuggee =>
+          val jdiPort = debuggee.readJdiPort()
+          val debugger = Debugger(jdiPort, expressionEvaluator, maxDuration/* , verbose = true */)
```

Comment:
What does the `/* , verbose = true */` mean? I would expect nothing, or `, /* verbose = */ true)`.

Reply:
Again, I left this to facilitate debugging. When a test fails it is often useful to print the execution of the debug steps to see where and why it failed.

Reply:
added comment